packages/sdk/src/types/runtime/appearance.ts

```diff
@@ -19,13 +20,22 @@ export class Appearance implements AppearanceLike {
     private _materialId = ZeroGuid;
     // tslint:enable:variable-name
 
-    /** Whether the actor with this appearance should be visible */
+    /**
+     * This actor's visibility preference, independent of its parent. See
+     * [[activeAndEnabled]] for the actual visibility state.
+     */
     public enabled = true;
 
+    /** Whether this actor is visible */
     public get activeAndEnabled() {
         return (!this.actor.parent || this.actor.parent.appearance.enabled) && this.enabled;
     }
 
+    /**
+     * Display this actor only for members of the contained groups.
+     */
+    public readonly enabledFor = new UserGroup();
```

Comment:
Most appearance objects won't have group affinities, so this can be lazily allocated.